src/yaml/pull_parser.cr

```diff
@@ -177,12 +235,9 @@ class YAML::PullParser
   end
 
   def read_null_or
-    if kind == EventKind::SCALAR && (value = self.value).nil? || (value && value.empty?)
-      read_next
-      nil
-    else
-      yield
-    end
+    read_null
+  rescue ParseException
```

Comment:
Raising exceptions is actually quite slow so you might want to replicate `read_null` exactly but `s/raise/yield`.

Reply:
done!